builder/simplify.cc

```diff
@@ -321,59 +327,75 @@ class simplifier : public node_mutator {
     var src;
   };
   symbol_map<buffer_info> buffers;
-  bounds_map expr_bounds;
+  symbol_map<expr_info> expr_bounds;
 
-  interval_expr result_bounds;
+  expr_info result_bounds;
 
-  void set_result(expr e, interval_expr bounds) {
-    assert(!result_bounds.min.defined() && !result_bounds.max.defined());
+  void set_result(expr e, expr_info bounds) {
+    assert(!result_bounds.bounds.min.defined() && !result_bounds.bounds.max.defined());
     result_bounds = std::move(bounds);
     node_mutator::set_result(std::move(e));
   }
   void set_result(stmt s) {
-    assert(!result_bounds.min.defined() && !result_bounds.max.defined());
-    result_bounds = interval_expr();
+    assert(!result_bounds.bounds.min.defined() && !result_bounds.bounds.max.defined());
+    result_bounds = {interval_expr(), modulus_remainder()};
     node_mutator::set_result(std::move(s));
   }
   // Dummy for template code.
-  void set_result(stmt s, interval_expr) { set_result(std::move(s)); }
+  void set_result(stmt s, expr_info) { set_result(std::move(s)); }
 
 public:
   simplifier() {}
-  simplifier(const bounds_map& expr_bounds) : expr_bounds(expr_bounds) {}
+  simplifier(const bounds_map& bounds, const alignment_map& alignment) {
+    for (size_t ix = 0; ix < bounds.size(); ix++) {
```

Comment:
This could be one loop over `std::max(bounds.size(), alignment.size())`. Only do so if it's less/simpler code.